src/core.ts

```diff
@@ -96,37 +96,51 @@ function getExposedContent(ast: SourceUnit, inputPath: string, contractMap: Cont
         return [
           contractHeader.join(' '),
           spaceBetween(
+            // slots for storage function parameters
             ...getAllStorageArguments(externalizableFunctions).map(a => [
               `mapping(uint256 => ${a.storageType}) internal ${prefix}${a.storageVar};`,
             ]),
-            makeConstructor(c, contractMap),
-            ...getInternalVariables(c, contractMap).map(v => {
-              return [
-                [
-                  'function',
-                  `${prefix}${v.name}(${getVarGetterArgs(v).map(a => `${a.type} ${a.name}`).join(', ')})`,
-                  'external',
-                  'view',
-                  'returns',
-                  `(${getVarGetterReturnType(v)})`,
-                  '{'
-                ].join(' '),
-                [
-                  `return ${v.name}${getVarGetterArgs(v).map(a => `[${a.name}]`).join('')};`,
-                ],
-                '}',
-              ];
+            // events for internal returns
+
+            ...externalizableFunctions.filter(fn => isNonPayableWithReturns(fn)).map(fn => {
+              const fnName = clashingFunctions[getFunctionId(fn)] === 1 ? fn.name : getFunctionNameStorageQualified(fn);
+              const fnArgs = getFunctionArguments(fn);
+              const evName = `${prefix}${[ fnName, ...fnArgs.map(a => a.type.split(' ').slice(0, 1).join('').replace(/[\.\[\]]/g, '_')), 'ReturnEvent' ].join('_')}`;
```

Comment:
Would suggest putting this in a helper since it's shared by the event definitions and the function definitions.